core/src/main/java/org/apache/calcite/plan/RelTrait.java

```diff
@@ -85,4 +87,14 @@
    * @param planner Planner
    */
   void register(RelOptPlanner planner);
+
+  /**
+   * Applies a mapping to this trait.
+   *
+   * @param mapping   Mapping
+   * @return trait with mapping applied
+   */
+  default <T extends RelTrait> T apply(Mappings.TargetMapping mapping) {
```

Comment:
We apply here only `Mappings.TargetMapping`. Can we apply other types of mappings?

Reply:
Can I have your use cases?

Reply:
I don't have particular use-cases. To be honest, I don't know much about mappings, so I'm just wondering if other mapping types are applicable here.

Reply:
Hmm, I will think about it. Thanks.

Reply:
Hi @rkondakov, I know the mappings is confusing. But I searched around, didn't find any uses other than TargetMapping. I will leave it TargetMapping as it is, but we can add other once we found there is a need.